src/library.js

```diff
@@ -988,7 +988,7 @@ LibraryManager.library = {
       var info = EXCEPTIONS.infos[ptr];
       assert(info.refcount > 0);
       info.refcount--;
-      if (info.refcount === 0) {
+      if (info.refcount === 0 && ptr !== EXCEPTIONS.last) {
```

Comment:
this change looks suspicious to me. how would the last exception get cleaned up, if not here? (trying to decref it later would fail as the refcount wouldn't be positive, as we assert a few lines above)